### PR TITLE
*: bump github.com/knz/go-libedit

### LINKED ISSUE #34059: teamcity: failed test: TestLint
The following tests appear to have failed on master (lint): TestLint/TestVet, TestLint

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestLint).

[#1096909](https://teamcity.cockroachdb.com/viewLog.html?buildId=1096909):

```
TestLint/TestVet
--- FAIL: lint/TestLint: TestLint/TestVet (167.940s)

------- Stdout: -------
=== PAUSE TestLint/TestVet
        lint_test.go:1087: 
            In file included from wrap-common.c:1:
        lint_test.go:1087: 
            In file included from vendor/github.com/knz/go-libedit/unix/src/libedit-common.c:1:
        lint_test.go:1087: 
            vendor/github.com/knz/go-libedit/unix/src/c-libedit/common.c:371:21: warning: passing 'wint_t *' (aka 'unsigned int *') to parameter of type 'wchar_t *' (aka 'int *') converts between pointers to integer types with different sign [-Wpointer-sign]
        lint_test.go:1087: 
            vendor/github.com/knz/go-libedit/unix/src/c-libedit/histedit.h:267:37: note: passing argument to parameter here
        lint_test.go:1087: 
            In file included from wrap-search.c:1:
        lint_test.go:1087: 
            In file included from vendor/github.com/knz/go-libedit/unix/src/libedit-search.c:1:
        lint_test.go:1087: 
            vendor/github.com/knz/go-libedit/unix/src/c-libedit/search.c:607:20: warning: passing 'wint_t *' (aka 'unsigned int *') to parameter of type 'wchar_t *' (aka 'int *') converts between pointers to integer types with different sign [-Wpointer-sign]
        lint_test.go:1087: 
            vendor/github.com/knz/go-libedit/unix/src/c-libedit/histedit.h:267:37: note: passing argument to parameter here



TestLint
--- FAIL: lint/TestLint (191.870s)





```

Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by petermattis**
This picks up a change which silences the pointer-sign conversion warnings which were in turn causing `TestLint/TestVet` to error spuriously.

Fixes #34059

Release note: None

### CODE REVIEW

**Review by jordanlewis — APPROVED** [review]

**Comment by petermattis** [comment]
Ugh, I can't whack vendor like this without the upstream change (we check that the contents of vendor match `Gopkg.lock`). I can't merge the upstream change without @knz's approval: https://github.com/knz/go-libedit/pull/15.

**Comment by petermattis** [comment]
I updated this PR to bump the version of `github.com/knz/go-libedit` we're using, and to unskip `TestLint/TestVet`. PTAL.

**Review by knz — APPROVED** [review]

**Comment by petermattis** [comment]
bors r=jordanlewis,knz

**Pull request merged**
